서버(백엔드) 초기 설정 및 모델링

shythm opened the pull request:
Closes #3 

## 변경 사항
- fastify 기반 서버 애플리케이션 구축
- swagger 연동
- 테스트용 ping/pong API 추가
- 모델, 레포지토리 레이어, 서비스 레이어 초기 인터페이스 추가
  - `server/core/model.ts`, `server/core/repositories.ts`, `server/core/services.ts`
  - 추후 레포지토리 구현체들은 `server/repositories` 디렉터리 밑에, 서비스 구현체들은 `server/services` 디렉터리 밑에 추가할 예정

shythm committed:
setup: server 초기 설정 및 환경 구축

shythm committed:
feat: 초기 모델, 레포지토리 레이어, 서비스 레이어 인터페이스 추가

DrCloy commented on server/src/core/models.ts:
```diff
@@ -0,0 +1,98 @@
+  stopwatchId: string | null; // 스톱워치(계수기) ID
+}
+
+export interface Participant {
+  id: string;
+  divisionId: string; // 대회 부문 ID
+  name: string; // 경연자 이름
+  teamName: string; // 팀 이름
+  robotName: string; // 로봇 이름
+  comment: string; // 하고 싶은 말
+  orderRaw: number; // 원본 경연 순번
+  givenTime: number; // 주어진 시간(ms)
+  createdAt: Date; // 생성 시각
+}
```
특정 위치에선 Participant가 경연자로, 다른 위치에선 참가자로 comment가 달려 있네요. 의미 상 혼돈을 막기 위해 하나로 통일하는 것이 좋을 거 같습니다.

DrCloy commented on server/src/core/models.ts:
```diff
@@ -0,0 +1,98 @@
+   * - "stopwatch": 계수기 H/W로부터 자동으로 기록된 값
+   * - "manual": 관리자가 수동으로 입력한 값
+   */
+  source: "stopwatch" | "manual";
```
기록의 source 항목에서 manual로 되어 있으면 이 기록이 수동 계수를 통해 생성된 기록인지, 관리자가 직접 값을 입력하여 생성된 기록인지 구별할 수 없을 거 같습니다. 두 항목을 구별할 수 있게 추가적인 값이 필요할 거 같습니다.

DrCloy commented on server/src/core/models.ts:
```diff
@@ -0,0 +1,98 @@
+    records: Record[]; // 현재 경연자의 기록 목록
+    manualRecords: ManualRecord[]; // 현재 경연자의 수동 계수 기록 목록
+  } | null;
+  nextRunner: Participant | null; // 다음 경연자 정보
```
일반적으로 다음 경연자를 보여줄 때 3~5명 정도 목록으로 보여주니 `nextRunnder` 항목도 `Participant`의 배열이 되는 것이 좋지 않을까요?

DrCloy commented on server/src/core/services.ts:
```diff
@@ -0,0 +1,399 @@
+   * 특정 대회의 그 아래의 부문들을 조회할 수 있다.
+   */
+  getCompetition(
+    actorId: string,
```
`actorId`가 어떤걸 의미하는지 잘 모르겠습니다. 추측상 클라이언트를 접근할 때 로그인 한 계정의 Id로, 권한 인증을 위해 필요한 것으로 보이긴 한데 정확히 어떤 용도인지 잘 이해되지 않습니다.

DrCloy reviewed (changes requested):
수정 부탁합니다

shythm committed:
feat: Record.source에 "other" 추가, Record.nextRunner를 nextRunners로 변경

- 주석 오기 수정
- 기타 다른 방법으로 기록된 값과 수동 계수 기록으로 취합된 값을 구분할 수 있도록 "other" Record.source 추가
- Progress.nextRunner를 nextRunners로 변경하여 최대 5명까지의 경연자 정보를 담을 수 있도록 배열로 둠

DrCloy reviewed (approved):
LGTM
